activerecord/lib/active_record/errors.rb

```diff
@@ -353,20 +353,24 @@ class Deadlocked < TransactionRollbackError
   class IrreversibleOrderError < ActiveRecordError
   end
 
+  # Superclass for all timeout-related errors
+  class Timeout < StatementInvalid
+  end
+
   # LockWaitTimeout will be raised when lock wait timeout exceeded.
-  class LockWaitTimeout < StatementInvalid
+  class LockWaitTimeout < Timeout
   end
 
   # StatementTimeout will be raised when statement timeout exceeded.
-  class StatementTimeout < StatementInvalid
+  class StatementTimeout < Timeout
   end
 
   # QueryCanceled will be raised when canceling statement due to user request.
-  class QueryCanceled < StatementInvalid
+  class QueryCanceled < Timeout
```

Comment:
I don't think we can consider this a Timeout. It is at user request so it is not timeout by definition.

Reply:
`QueryCanceled` doesn't happen on a user request. It happens when MySQL server kills the query on its end, for instance with [`pt-kill`](https://www.percona.com/doc/percona-toolkit/LATEST/pt-kill.html).

Reply:
pt-kill is an user requested canceling. The user need to send a `kill` statement to the mysql server. it just happens that `pt-kill` does that automatically for you.